### PR TITLE
Replace our custom OnOffSwitch with PatternFly-React Switch

### PR DESCRIPTION
**Pull request by martinpitt**
Drop our own component, it's not being used any more. Also drop the
corresponding Playground demo, the upstream docs at
https://patternfly-react.surge.sh/patternfly-3/ are a much more
comprehensive place for that.

The integration tests that cover functionality in RHEL 7 Base need to
understand both variants, so introduce some helper functions.

 - [ ] Don't react to middle/right mouse clicks: https://github.com/Julusian/react-bootstrap-switch/issues/53
 - [ ] Fix wrong "overlapping" design when enabling on main page and switching to firewall page, changing on main page, and switching back to firewall page (i. e. React component is around while main page is visible)
 - [ ] ~Switching on firewall sometimes gets stuck with disabled switch~ - already in master, this is PR #9125
 - [ ] When the VDO details page appears, the switches animate from On to Off gratuitously.

### CODE REVIEW

**Comment by martinpitt** [comment]
Pushed adjustments to tests.

Locally I still get a failure in `check-packagekit TestAutoUpdates.testInstall`, which I need to debug. But let's see what else still fails.

**Comment by martinpitt** [comment]
@larskarlitski , @Gundersanne , @mvollmer: Not sure who I should ask to review this one. Any volunteers? :-)

File: pkg/storaged/storage-controls.jsx [hunk 1]
```diff
@@ -197,11 +197,11 @@ class StorageOnOff extends React.Component {
         return (
             <StorageControl excuse={this.props.excuse}
                             content={(excuse) => (
-                                <OnOffSwitch state={this.state.promise
+                                <Switch value={this.state.promise
                                     ? this.state.promise_goal_state
                                     : this.props.state}
-                                                 enabled={!excuse && !this.state.promise}
-                                                 onChange={onChange} />
+                                        disabled={excuse || this.state.promise}
```
Comment (mvollmer) [inline pkg/storaged/storage-controls.jsx]: This needs to be a real boolean or React will complain: `Boolean(excuse || )`

**Comment by mvollmer** [comment]
When zoomed to 120%, ther seems to be some rounding error and the "On" part of the switch doesn't completely disappear:
onoff-rounding

**Comment by mvollmer** [comment]
If the Firewallpage is loaded while the firewall is off, switching it on will move the switch, but it will stay disabled.

**Comment by mvollmer** [comment]
When a page with a Switch is loaded and the Switch should be initially off, there sometimes is a gratuitous animation from On to Off.

File: pkg/storaged/storage-controls.jsx (hunk 1 above)
Comment (martinpitt) [reply pkg/storaged/storage-controls.jsx]: Thanks, fixed (not pushed yet).

**Comment by martinpitt** [comment]
@mvollmer: Thanks for testing: I cannot reproduce the pixel error on zooming and the gratuitous "off" animation, but I was able to reproduce the other three. I made TODO checkboxes for them.

**Comment by martinpitt** [comment]
> If the Firewallpage is loaded while the firewall is off, switching it on will move the switch, but it will stay disabled.

This is happening with master, too. This is a rather tough bug which @larskarlitski tried to fix in PR #9125. When enabling the service, `onFirewallChanged` never gets called from our service.js helper.

So I crossed that out, it's unrelated.

**Comment by martinpitt** [comment]
I reported the middle/right click action: https://github.com/patternfly/patternfly-react/issues/851

This is a bit awkward to fix from our side, so if you think that is an important enough issue, let's block this PR on fixing that.

**Comment by martinpitt** [comment]
The weird "overlapped" state happens if the firewall page is already loaded and active in the background, and the update happens then. The properties passed to `onFirewallChanged()` and `onSwitchChanged()` all seem to be correct. So this is another bug in PF-React, which I'll need to file. But this requires a non-trivial reproducer.

**Comment by mvollmer** [comment]
> The weird "overlapped" state happens if the firewall page is already loaded and active in the background, and the update happens then. 

This is likely a bug in react-bootstrap-switch as well: The switch looks weird because its elements are not the right size.  They probably get resized by JavaScript, and if that happens while the switch is invisible, this goes wrong. Invisible elements have width/height of zero.

**Comment by mvollmer** [comment]
> When a page with a Switch is loaded and the Switch should be initially off, there sometimes is a gratuitous animation from On to Off.

This triggers it reliably for me: Go to the Storage main page, click on a VDO device that has Compression off. When the details page appears, the animation happens every time.

**Comment by mvollmer** [comment]
Pixel rounding issue: https://github.com/Julusian/react-bootstrap-switch/issues/54

**Comment by mvollmer** [comment]
> When a page with a Switch is loaded and the Switch should be initially off, there sometimes is a gratuitous animation from On to Off.

This is caused by the same issue as the funny layout: The Switch component measures itself, but when that is done while it is invisible, everything goes wrong.

We can work around this by providing a fixed size with `handleWidth={33}`, but the measure feature is actually nice for translations, for course.

We can also try to stop instantiating switches while they are invisible.  There aren't super many, so I'll have a look.

**Comment by mvollmer** [comment]
> We can also try to stop instantiating switches while they are invisible.

I think this is quite hard to do since pages are in iframes and the shell controls the visibility of those.  The shell could use `visibility: 0%` instead of `display: none`, maybe.

**Comment by mvollmer** [comment]
The Switch will remeasure itself every time its properties change.  So if, say, the Storage page has been loaded but is currently not visible (because the user has switched to the Networking page, say), and then something in the storage model changes, the Swicthes will remeasure themselves while being invisble, and will get it all wrong.

**Comment by mvollmer** [comment]
> The Switch will remeasure itself every time its properties change.

This is the reason for the funny layout of the Firewall switch.  Loading the page is fine, but when the switch changes value while the iframe for the page is invisible, we end up with the funny layout.

**Comment by mvollmer** [comment]
I think the "layout breaks when not displayed" issue is a deal breaker for us.  It should be possible to change react-bootstrap-switch itself to not need any measurements, I think... something for a rainy sunday afternoon.

**Comment by martinpitt** [comment]
Yes, I agree, this is too broken right now. Let's shelve this until react-bootstrap-switch is sufficiently fixed.

**Pull request closed without merging**